libnavigation-ui/src/main/java/com/mapbox/navigation/ui/camera/SimpleCamera.kt

```diff
@@ -30,20 +35,61 @@ open class SimpleCamera : Camera() {
         return DEFAULT_ZOOM
     }
 
-    override fun overview(routeInformation: RouteInformation): List<Point> =
-        getRoute(routeInformation)?.run {
-            generateRouteCoordinates(this)
+    /**
+     * If the [RouteInformation] does not include a [RouteProgress] instance the points
+     * for the entire route will be returned. If a [RouteProgress] is present, the distance
+     * traveled will be used to truncate the route points such that the final collection of points
+     * returned will contain the point nearest the distance traveled until the end of the route.
+     */
+    override fun overview(routeInformation: RouteInformation): List<Point> {
+        return getRoute(routeInformation)?.run {
+            when (routeInformation.routeProgress) {
+                null -> generateRouteCoordinates(this)
+                else -> getCoordinates(this, routeInformation.routeProgress)
+            }
         } ?: emptyList()
+    }
 
     private fun getRoute(routeInformation: RouteInformation): DirectionsRoute? =
         when (routeInformation.route) {
             null -> routeInformation.routeProgress?.route
             else -> routeInformation.route
         }
 
-    private fun generateRouteCoordinates(route: DirectionsRoute?): List<Point> =
-        route?.geometry()?.let { geometry ->
-            val lineString = LineString.fromPolyline(geometry, Constants.PRECISION_6)
-            lineString.coordinates()
+    private fun getCoordinates(
+        route: DirectionsRoute?,
+        routeProgress: RouteProgress?
+    ): List<Point> {
+        return ifNonNull(
+            route,
+            route?.geometry(),
+            routeProgress
+        ) { theRoute, _, theRouteProgress ->
+            val lineString = getLineString(theRoute)!!
```

Comment:
nit: This force cast can be cleaned up.

Reply:
I pushed up a revision but you got to it first.